Move Dom Renderer Out Of Core

vsavkin opened the pull request.

IgorMinar commented on modules/angular2/src/core/dom/browser_adapter.dart:
```diff
@@ -1,520 +1,4 @@
-library angular.core.facade.dom;
+@deprecated
```
can we use `@Deprecated('Import this library from XXX instead')` instead?

IgorMinar replied:
throughout this PR...

vsavkin replied:
Fixed

IgorMinar commented on modules/angular2/test/public_api_spec.ts:
```diff
@@ -479,6 +479,11 @@ var NG_ALL = [
+  'DomAdapter.attrToPropMap:dart',
+  'DomAdapter.attrToPropMap=:dart',
+  'setRootDomAdapter()',
+  'DOM:dart',
```
Why is this DART only?

IgorMinar replied:
from code it looks like it's used in both ts and dart

vsavkin replied:
Fixed.

IgorMinar commented on modules/angular2/test/public_api_spec.ts:
```diff
@@ -479,6 +479,11 @@ var NG_ALL = [
   'DirectiveResolver',
   'DirectiveResolver.resolve()',
   'DynamicComponentLoader',
+  'DomAdapter',
```
This is a new reexport, right?

vsavkin replied:
Yes. And this is public API of core.

IgorMinar commented on tools/broccoli/broccoli-check-imports.ts:
```diff
@@ -0,0 +1,88 @@
+  rebuild(treeDiff: DiffResult) {
+    const errors = this.checkAllPaths(treeDiff);
+    if (errors.length > 0) {
+      throw new Error(`The following imports are not allowed:\n${errors.join("\n")}`);
```
".... because they break barrel boundaries:" - we should explain why

vsavkin replied:
Fixed

IgorMinar commented on tools/broccoli/broccoli-check-imports.ts:
```diff
@@ -0,0 +1,88 @@
+    if (errors.length > 0) {
+      throw new Error(`The following imports are not allowed:\n${errors.join("\n")}`);
+    }
+    this.symlinkInputAndCache();
```
I believe that you need to do this only once - during the `initRun`

vsavkin replied:
It is done only once. I renamed the function to make it more clear.

IgorMinar commented on tools/broccoli/broccoli-check-imports.ts:
```diff
@@ -0,0 +1,88 @@
+
+  private checkFilePath(filePath, errors: string[]) {
+    const sourceFilePath = path.join(this.inputPath, filePath);
+    // change this change to say ends with
```
huh?

vsavkin replied:
It was an informal todo that I already fixed.

IgorMinar commented on tools/broccoli/broccoli-check-imports.ts:
```diff
@@ -0,0 +1,88 @@
+  private checkFilePath(filePath, errors: string[]) {
+    const sourceFilePath = path.join(this.inputPath, filePath);
+    // change this change to say ends with
+    if (endsWith(sourceFilePath, ".ts") && fs.existsSync(sourceFilePath)) {
```
this will match `d.ts` files. is that ok?

vsavkin replied:
I think the rules should be the same for ts and d.ts files.

IgorMinar commented on tools/broccoli/broccoli-check-imports.ts:
```diff
@@ -0,0 +1,88 @@
+      lines.forEach(line => {
+        const res = CheckImports.IMPORT_REGEXP.exec(line);
+        if (res && !this.isAllowedImport(filePath, res[1])) {
+          errors.push(`${filePath}: '${res[1]}'`);
```
can we make the difference between filePath and import more clear in the output? currently the only diff is that the first part has `.ts` extension.

vsavkin replied:
fixed

IgorMinar commented on tools/broccoli/broccoli-check-imports.ts:
```diff
@@ -0,0 +1,88 @@
+  private checkAllPaths(treeDiff: DiffResult) {
+    const errors = [];
+    const changesFiles = treeDiff.addedPaths.concat(treeDiff.changedPaths);
+    changesFiles.forEach(_ => this.checkFilePath(_, errors));
```
it's kind of ugly how you pass in the errors array, just return a new array and concat it here.

vsavkin replied:
Fixed it. The work on core made my paranoid about creating extra arrays.

IgorMinar commented:
minor comments... please resolve and merge

vsavkin committed:
refactor(core): move render/dom from core

Currently, core depends on DomRenderer, which depends on the browser.
This means that if you depend on angular2/core, you will always
pull in the browser dom adapter and the browser render, regardless
if you need them or not.

This PR moves the browser dom adapter and the browser renderer out of core.

BREAKING CHANGE

If you import browser adapter or dom renderer directly (not via angular2/core),
you will have to change the import path.
